mlcomp/parallelm/pipeline/dag_node.py

```diff
@@ -87,6 +89,11 @@ def _update_default_values(self, input_args):
         comp_args = self._comp_desc[json_fields.COMPONENT_DESC_ARGUMENTS]
         for comp_arg in comp_args:
             arg_key = comp_arg[json_fields.COMPONENT_DESC_ARGUMENT_KEY]
+            # check if argument has an option to get value from env
+            if arg_key not in input_args and json_fields.COMPONENT_DESC_ARGUMENT_ENV_VAR in comp_arg:
```

Comment:
You may simplify the two `if` conditions and make it clearer as follow (keeping the same logic):
```
if arg_key not in input_args:
      if json_fields.COMPONENT_DESC_ARGUMENT_DEFAULT_VAL in comp_arg:
            input_args[arg_key] = comp_arg[json_fields.COMPONENT_DESC_ARGUMENT_DEFAULT_VAL]
      elif json_fields.COMPONENT_DESC_ARGUMENT_ENV_VAR in comp_arg:
            env_var_value = os.environ.get(comp_arg[json_fields.COMPONENT_DESC_ARGUMENT_ENV_VAR])
            if env_var_value:
                 input_args[arg_key] = env_var_value
```

Reply:
Yes, I thought about it, but...
Env Var should have priority over default.
```
if arg_key not in input_args:
      if json_fields.COMPONENT_DESC_ARGUMENT_ENV_VAR in comp_arg:
            env_var_value = os.environ.get(comp_arg[json_fields.COMPONENT_DESC_ARGUMENT_ENV_VAR])
            if env_var_value:
                 input_args[arg_key] = env_var_value
     elif json_fields.COMPONENT_DESC_ARGUMENT_DEFAULT_VAL in comp_arg:
            input_args[arg_key] = comp_arg[json_fields.COMPONENT_DESC_ARGUMENT_DEFAULT_VAL]
```

But in tis case, if envVar is defined in attribute, but no actual environment var was set, value won't be added into attributes, and also elif clause won't be triggered for default. That's why I haven't put these conditions under nested if.

On other hand, default doesn't really make sense in this case, where we want to

Reply:
Your more general case means that the default value has a priority over the default value, because it overwrites it.
You can just read the environment variable this way:
```env_var_value = os.environ.get(comp_arg[json_fields.COMPONENT_DESC_ARGUMENT_ENV_VAR], None)```
(Notice the ```None``` value).
and then follow the if else rules.